resources/sql/queries.sql

```diff
@@ -71,16 +74,23 @@ UPDATE catalogue_item
 SET organization = :organization
 WHERE id = :id;
 
+-- :name set-catalogue-item-data! :!
+UPDATE catalogue_item
+SET catalogueitemdata = :catalogueitemdata::jsonb
+WHERE id = :id;
+
 -- :name create-catalogue-item! :insert
 -- :doc Create a single catalogue item
 INSERT INTO catalogue_item
-(formid, resid, wfid, organization, enabled, archived, start)
+(formid, resid, wfid, organization, enabled, archived, start, catalogueitemdata)
 VALUES (:form, :resid, :wfid, :organization,
 --~ (if (contains? params :enabled) ":enabled" "true")
 ,
 --~ (if (contains? params :archived) ":archived" "false")
 ,
 --~ (if (contains? params :start) ":start" "now()")
+,
+/*~ (if (contains? params :catalogueitemdata) */ :catalogueitemdata::jsonb /*~*/ NULL /*~ ) ~*/
```

Comment:
found this example from another row in the same file for using `NULL` as default value, which is why it looks different from the other default values above

Reply:
i think it's possible to pass `:catalogueitemdata nil` directly to the db function which would be considered ok by `contains?` as it only checks that the key exists.

in normal scenario this would not happen though, as the service layer call ensures that any value is properly jsonified